pymatviz/ptable.py

```diff
@@ -703,293 +766,292 @@ def histogram(
+            overwrite_colors: dict[str, ColorType] | None = None,
+            **kwargs: dict[str, Any],
+        ) -> None:
+            """Init Heatmap plotter.
+
+            Args:
+                values_show_mode ("value" | "fraction" | "percent" | "off"):
+                    Values display mode.
+                tile_colors (dict[str, ColorType] | None): Tile colors.
+                    Defaults to None for auto generation.
+                overwrite_colors (dict[str, ColorType] | None): Optional
+                    overwrite colors. Defaults to None.
+                kwargs (dict): Kwargs to pass to super class.
+            """
+            super().__init__(**kwargs)
+
+            self.values_show_mode = values_show_mode
+
+            # Normalize data for "fraction/percent" modes
+            if values_show_mode in {"fraction", "percent"}:
+                self.data = normalize_data(self.data, percentage=False)
+
+            self.overwrite_colors = overwrite_colors or {}
+            self.tile_colors = tile_colors
+
+        @property
+        def tile_colors(self) -> dict[str, ColorType]:
+            return self._tile_colors
+
+        @tile_colors.setter
+        def tile_colors(
+            self,
+            tile_colors: dict[str, ColorType] | None = None,
+        ) -> None:
+            """An element symbol to color mapping.
+
+            Args:
+                tile_colors (dict[str, ColorType] | None): Tile colors.
+                    Defaults to None for auto generation.
+            """
+            # Generate tile colors from values if not given
+            self._tile_colors = tile_colors or {}
+
+            if tile_colors is None:
+                for symbol in self.data.index:
+                    # Get value and map to color
+                    value = self.data.loc[symbol, Key.heat_val][0]
+
+                    self._tile_colors[symbol] = self.cmap(self.norm(value))
+
+            # Overwrite colors if any
+            self._tile_colors |= self.overwrite_colors
+
+        def add_elem_values(
+            self,
+            *,
+            text_fmt: str,
+            pos: tuple[float, float] = (0.5, 0.25),
+            text_color: str = "AUTO",
+            kwargs: dict[str, Any] | None = None,
+        ) -> None:
+            """Format and show element values.
+
+            Args:
+                text_fmt (str): f-string format for the value text.
+                pos (tuple[float, float]): Position of the value in the tile.
+                text_color (str): Value text color.
+                kwargs (dict): Additional keyword arguments to pass to the `ax.text`.
+            """
+            # Update symbol kwargs
+            kwargs = kwargs or {}
+            kwargs.setdefault("fontsize", 12)
+
+            # Generate values format depending on the display mode
+            if text_fmt == "AUTO":
+                text_fmt = ".1%" if values_show_mode == "percent" else ".3g"
```

Comment:
Here we have some inconsistency, especially with the `normalize_data` util, as `text_fmt` as `".1%"` would multiply data by 100. Perhaps instead of taking the format f-string, we ask for the total number of digits?

Reply:
i really like the f-string float format syntax because its so versatile. has way more features than just precision like left/right alignment, zero padding, thousand separator, percentages, etc. so would always piggy back on that where possible over restricting user control to only precision

Reply:
Okay sure no problem :) Let's use f-string. That's always a hard choice, where more flexible means more possibilities to trigger expected behavior :)